spec/spec_helper.rb

```diff
@@ -48,6 +77,29 @@ def build_event(message)
 
 RSpec.configure do |c|
   c.include GrokHelpers
+  c.include GrokHelpers::Shared
+  c.extend GrokHelpers::Shared
+end
+
+def describe_pattern(name, pattern_modes = [ nil ], &block)
```

Comment:
I think this method is unnecessarily forgiving about the arguments it accepts, since all current callers send a second array argument we can remove the default value assignment.

```suggestion
/**
 * @param name [String]
 * @param pattern_modes = [Array[String]]
 */
def describe_pattern(name, pattern_modes, &block)
```

Reply:
this might have been misleading - the idea was to still use `describe_pattern` for stuff that is "left as is",
in which case the plugin just runs in legacy mode (die the `nil` pattern mode). 
updated sample specs here for the `REDISMONLOG` to use `describe_pattern` and thus re-use the grok setup - we could go be explicit about the `legacy` mode there if you have a preference towards explicitness.

have simply anticipated that we might want to keep a lot of specs as is while still re-using pattern/message/grok setup.

Reply:
@yaauie unless you really want us to be explicit I would keep the `pattern_modes` implicit default.
since the plugin is expected to boot in legacy mode, for now ...